sklearn/neighbors/kde.py

```diff
@@ -123,6 +127,11 @@ def fit(self, X, y=None):
         """
         algorithm = self._choose_algorithm(self.algorithm, self.metric)
         X = check_array(X, order='C', dtype=DTYPE)
+        
+        if self.bandwidth == 'scott':
+            self.bandwidth = 3.5*X.std()/len(X)**0.3333
```

Comment:
Instead of overwriting the parameter, please introduce an estimated attribute `bandwidth_`.

Numpy's histogramming function implements this with higher precision constants ((24.0 * np.pi**0.5) ** 1/3. instead of 3.5; 1/3. instead of 0.3333). Do we care? Should we use higher precision constants here?

Also, I'd prefer spacing like `3.5 * X.std() / len(X)**0.3333` which emphasises the order of operations, while the spaces add readability.

Reply:
Just to clarify, `bandwidth_` should contain the string defining the method right? The `bandwidth` attribute is used as a float later in other functions and I'd rather not change those functions too.

I'll change it to the higher precision constants for consistency, in practice this will have little effect.

Reply:
Can I ask why you would even use `3.5` instead of `3.49` as in [Scott's paper](http://www.stat.cmu.edu/~rnugent/PCMI2016/papers/ScottBandwidth.pdf) (5. DATA-BASED HISTOGRAMS, p.608)?
By the same token, it is true Python is not very good at high-precision calculations but why don't you just let python deal itself with the `1./3.` fraction instead of plugging `0.333` or `0.3333` or etc... ?
Apologies if I'm totally missing the point.